### PR TITLE
rpc: getblock/getrawtransaction/decode*/gettxout fixups

### PR DESCRIPTION
**Pull request by jonatack**
In draft to re-verify the changes and see Drahtbot-enabled crossover with other pulls.

### CODE REVIEW

**Comment by jonatack** [comment]
Hm, just a few hours later and now this needs to be redone :D

File: src/rpc/rawtransaction.cpp [hunk 1]
```diff
@@ -411,10 +411,10 @@ static RPCHelpMan decodescript()
                  "Result of a witness script public key wrapping this redeem script (not returned for types that should not be wrapped)",
                  {
                      {RPCResult::Type::STR, "asm", "String representation of the script public key"},
+                     {RPCResult::Type::STR, "desc", "Inferred descriptor for the script"},
                      {RPCResult::Type::STR_HEX, "hex", "Hex string of the script public key"},
-                     {RPCResult::Type::STR, "type", "The type of the script public key (e.g. witness_v0_keyhash or witness_v0_scripthash)"},
                      {RPCResult::Type::STR, "address", /*optional=*/true, "The Bitcoin address (only if a well-defined address exists)"},
-                     {RPCResult::Type::STR, "desc", "Inferred descriptor for the script"},
+                     {RPCResult::Type::STR, "type", "The type (one of: " + GetAllOutputTypes() + ")"},
```
Comment (luke-jr) [inline src/rpc/rawtransaction.cpp:417]: I think " of the script public key" is relevant here

**Comment by fanquake** [comment]
> In draft to re-verify the changes and see Drahtbot-enabled crossover with other pulls.

What's the status of this? Have the changes been re-verified? Drahtbot has reported the conflicts at least one time.

Looking at the changes, it seems they could be consolidated. Commits like 96b3a9b72934e62c429a29dc85ceccbf75a662dd and 68a041dd12b3bfee1f8308cd3586dccd3efefed4 are doing the same refactor of changing `"The type, eg 'pubkeyhash'"` to `"The type (one of: " + GetAllOutputTypes() + ")"` (also in other commits). It probably makes sense to do that all in a single change, rather than spread across 5 commits.

**Comment by jonatack** [comment]
Yes, this started out as fixes for one RPC, then I saw similar ones across others, and asked myself whether people prefer changes by kind or by RPC, which is another reason it was in draft. I plan to go through this with fresh eyes and either break it down to changes for one RPC or for one kind of change in one or more small, easy-to-review pulls so they don't just sit there for months.

**Comment by fanquake** [comment]
> and either break it down to changes for one RPC or for one kind of change in one or more small

I don't think this needs to be made any more granular. What is the motivation for per-RPC commits?

> easy-to-review pulls so they don't just sit there for months.

If you're referencing this PR; it's been in draft since it was opened, and needed rebase for ~ 1 month. So it's not super surprising that it hasn't had much interest / review.

**Pull request closed without merging**